### PR TITLE
Feature/refactor login view

### LINKED ISSUE #1082: Improve Code Legibility in login_view
The widgets are deeply nested and could be declared in other files.

### PR DESCRIPTION
**Pull request by Spoodern0t**
### ⁉️ Related Issue
Closes #1082 

## 📖 Description
Moved several widgets from login_view.dart to their own files in order to increase legibility.
Said new files are located at features/welcome/widgets

### 🧪 How Has This Been Tested?
This was tested by trying multiple login attempts (valid and invalid) and making sure that each widget responded the way it did before the change. 
The change does not affect any classes or files beyond login_view.dart, and the core function of the moved widgets was left intact.

### ☑️ Checklist before requesting a review
- [x ] I have performed a self-review of my code.
- [x ] If it is a core feature, I have added thorough tests.
- [x ] If needed, I added analytics.
- [x ] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`.

### CODE REVIEW

File: lib/features/welcome/login/login_view.dart
Comment (LouisPhilippeHeon) [inline lib/features/welcome/login/login_view.dart:1]: Tu peux retirer les lignes 124-131.

**Review by LouisPhilippeHeon — CHANGES_REQUESTED** [review]

File: lib/features/welcome/widgets/forgot_password.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import 'package:notredame/features/app/analytics/remote_config_service.dart';
+import 'package:notredame/features/app/integration/launch_url_service.dart';
+
+class ForgotPassword extends StatefulWidget{
+
+  const ForgotPassword(
+      {super.key});
+
+  @override
+  _ForgotPasswordState createState() => _ForgotPasswordState();
```
Comment (XavierPaquet-Rapold) [inline lib/features/welcome/widgets/forgot_password.dart:17]: Juste parce que c'est quelque chose qui va arriver dans une de mes PR:
`State<ForgotPassword> createState() => _ForgotPasswordState();`

File: lib/features/welcome/widgets/login_button.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  final ValueGetter<Future<String>> authenticate;
+
+  const LoginButton({
+    super.key,
+    required this.formKey,
+    required this.canSubmit,
+    required this.authenticate});
+
+  @override
+  _LoginButtonState createState() => _LoginButtonState();
```
Comment (XavierPaquet-Rapold) [inline lib/features/welcome/widgets/login_button.dart:22]: Même ici
```suggestion
  State<LoginButton> createState() => _LoginButtonState();
```

File: lib/features/welcome/widgets/login_footer.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:notredame/features/app/navigation/navigation_service.dart';
+import 'package:notredame/features/app/navigation/router_paths.dart';
+
+class LoginFooter extends StatefulWidget{
+
+  const LoginFooter(
+      {super.key});
+
+  @override
+  _LoginFooterState createState() => _LoginFooterState();
```
Comment (XavierPaquet-Rapold) [inline lib/features/welcome/widgets/login_footer.dart:18]: Même ici
```suggestion
  State<LoginFooter> createState() => _LoginFooterState();
```

File: lib/features/welcome/widgets/login_hero.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import 'package:notredame/utils/app_theme.dart';
+
+class LoginHero extends StatefulWidget{
+
+  const LoginHero(
+      {super.key});
+
+  @override
+  _LoginHeroState createState() => _LoginHeroState();
```
Comment (XavierPaquet-Rapold) [inline lib/features/welcome/widgets/login_hero.dart:12]: Même ici:
```suggestion
  State<LoginHero> createState() => _LoginHeroState();
```

File: lib/features/welcome/widgets/universal_code_text_field.dart [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  final String universalCode;
+
+  const UniversalCodeFormField({
+        super.key,
+        required this.validator,
+        required this.onEditionComplete,
+        required this.universalCode});
+
+  @override
+  _UniversalCodeFormFieldState createState() => _UniversalCodeFormFieldState();
```
Comment (XavierPaquet-Rapold) [inline lib/features/welcome/widgets/universal_code_text_field.dart:18]: Même ici:
```suggestion
  State<UniversalCodeFormField> createState() => _UniversalCodeFormFieldState();
```

**Review by XavierPaquet-Rapold — CHANGES_REQUESTED** [review]

File: lib/features/welcome/widgets/universal_code_text_field.dart (hunk 5 above)
Comment (LouisPhilippeHeon) [reply lib/features/welcome/widgets/universal_code_text_field.dart:18]: Dans le fichier pour le password text field il faudra penser à le changer aussi alors (ce fichier existe déjà et n'a pas été modifié)

**Review by XavierPaquet-Rapold — DISMISSED** [review]

**Review by LouisPhilippeHeon — APPROVED** [review]
LGTM

**Pull request merged**